File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/performancereport/PerformanceReport.kt
package com.ongo.domain.performancereport

import java.time.LocalDate
import java.time.LocalDateTime

data class PerformanceReport(
    val id: Long? = null,
    val userId: Long,
    val title: String,
    val period: String = "MONTHLY",
    val startDate: LocalDate,
    val endDate: LocalDate,
    val status: String = "DRAFT",
    val totalViews: Long = 0,
    val totalEngagement: Long = 0,
    val topVideoId: Long? = null,
    val topVideoTitle: String? = null,
    val reportUrl: String? = null,
    val createdAt: LocalDateTime? = null,
)

File: frontend/src/types/performanceReport.ts
export interface PerformanceReport {
  id: number
  title: string
  period: string
  startDate: string
  endDate: string
  status: 'DRAFT' | 'GENERATING' | 'COMPLETED' | 'SCHEDULED'
  totalViews: number
  totalEngagement: number
  topVideoId: number | null
  topVideoTitle: string | null
  reportUrl: string | null
  createdAt: string
}

export interface ReportSection {
  id: number
  reportId: number
  sectionType: string
  title: string
  content: string
  chartData: Record<string, unknown> | null
  sortOrder: number
}

export interface PerformanceReportSummary {
  totalReports: number
  scheduledReports: number
  avgViewsPerPeriod: number
  bestPeriod: string
  growthRate: number
}
